docs/design/self-worktree-delegation.md

```
@@ -0,0 +1,316 @@
+  branch: string,
+}
+```
+
+**Internal implementation**: Calls the existing worktree creation flow (`POST /api/repositories/:id/worktrees` equivalent) with `mode: 'prompt'` (or `mode: 'custom'` if branch is specified) and `autoStartSession: true`. Since the MCP endpoint runs inside the server process, it can call the service layer directly instead of going through HTTP.
+
+**Async handling**: Worktree creation is asynchronous (git operations, optional branch name generation). The MCP tool waits for the `worktree-creation-completed` internal event before returning the result to the agent. Timeout: 120 seconds.
+
+### 2. `get_session_status`
+
+Check the status of a delegated session.
+
```

markdownlint-cli2 (0.20.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)